syntax="proto3";

import "google/protobuf/timestamp.proto";
package usermanagement;


service oculusUserManagement {
   rpc addPlatform (platformRequest) returns (platformResponce) {}

  rpc register (registraionRequest) returns (registraionResponce) {}

  rpc getUser (userInfoRequest) returns (userInfoResponce) {}

  //Change after getting account setttings details
  rpc accountSettings (settingsRequest) returns (settingsResponce) {}

  rpc notifications (notificationRequest) returns (notificationResponce) {}
}


message platformRequest {
  string registrationId =1;
  string platform_name=2;
  Data data =3;
  string registration_time=10;
  string last_updated_time=11;
}
message Data{
    string appId=1;
    string accesstoken=2;
}


message platformResponce {
    string media_responce=1;
}

message registraionRequest {
  string username=1;
  FaceBook faceBook=5;
  LinkedIn linkedIn=6;
  Twitter twitter=7;
  Instagram instagram=8;
//  google.protobuf.Timestamp registrationTime=10;
//  google.protobuf.Timestamp lastUpatedTime=11;
}

message registraionResponce {
  string registrationId = 1;
}

message userInfoRequest{
  string registrationId=1;
}

message userInfoResponce{
  string userData=1;
}

message notificationRequest{

}

message notificationResponce{

}

message settingsRequest{

}

message settingsResponce{

}

message FaceBook{
 string accessToken=1;
 string appId=2;
}

message LinkedIn{
 string appId=1;
 string accessToken=2;
}

message Twitter{
    string bearerToken=1;
    string apiKey=2;
    string apiKeySecret=3;
    string accessToken=4;
    string accessTokenSecret=5;
}

message Instagram{
 string accessToken=1;
 string appId=2;
}
